crates/configuration/src/core.rs

```diff
@@ -3,7 +3,7 @@ use anyhow::anyhow;
 use serde_json::Number;
 use std::env;
 
-fn resolve_env_variables(config: serde_json::Value) -> anyhow::Result<serde_json::Value> {
+pub fn resolve_env_variables(config: serde_json::Value) -> anyhow::Result<serde_json::Value> {
```

Comment:
```suggestion
pub(crate) fn resolve_env_variables(config: serde_json::Value) -> anyhow::Result<serde_json::Value> {
```

If you need this in other modules `pub` is okay, but if it remains just for uses in this crate, let's not publish it.